api-controllers/ServiceBrokerApiController.js

```diff
@@ -134,36 +140,61 @@ class ServiceBrokerApiController extends FabrikBaseController {
       const previousPlan = _.find(plan.service.plans, ['id', previousPlanId]);
       return plan === previousPlan || _.includes(plan.manager.settings.update_predecessors, previousPlan.id);
     }
+    let lastOperationState = {
+      resourceGroup: plan.manager.resource_mappings.resource_group,
+      resourceType: plan.manager.resource_mappings.resource_type,
+      resourceId: req.params.instance_id,
+      start_state: CONST.APISERVER.RESOURCE_STATE.UPDATE,
+      started_at: new Date()
+    };
     return Promise
       .try(() => {
         if (!isUpdatePossible(params.previous_values.plan_id)) {
           throw new BadRequest(`Update to plan '${plan.name}' is not possible`);
         }
       })
       .then(() => {
-        return eventmesh.apiServerClient.patchResource({
-          resourceGroup: plan.resourceGroup,
-          resourceType: plan.resourceType,
-          resourceId: req.params.instance_id,
-          options: params,
-          status: {
-            state: CONST.APISERVER.RESOURCE_STATE.UPDATE,
-            lastOperation: {},
-            response: {}
-          }
-        });
-      })
-      .then(() => {
-        if (!plan.manager.async) {
-          return eventmesh.apiServerClient.getResourceOperationStatus({
+        serviceFlow = req._serviceFlow;
```

Comment:
just for my understanding , in case a bi-weekly / scheduled update comes along with the multi-az update , then how do we ensure that the deployment resource is locked ? Since bi-weekly update is not via CF and in the multi-az update we do not lock the deployment resource till the task is initiated , we can have collision right ?

Reply:
The lock resource is created with the resource id of the  instance. The lock resource has options attrib (LockResourceDetails) which is used by lock manager to poll for checking the status and holding the lock. Since the instance id is the resource ID of the lock resource and it wont be released till the serialworkflow resource turns to success/failed, scheduled update wont pass through. This is similar to how locking is done for RESTORE or Backup operation.

Reply:
Ok , got it . thanks for the info.